# Lecture 3e

## Interpretation of Categorical Variables

This example explores the interpreation of categorical varaibles as shifting the mean (intercept) of the reression

In [1]:
Credit <- read.csv("~/rotman/RSM8512/Data/Credit.csv") #load data
attach(Credit) #attach variable names


No documentation for ‘Credit’ in specified packages and libraries:
you could try ‘??Credit’

In [2]:
Credit

X   Income  Limit Rating Cards Age Education Gender Student Married
1    1   14.891  3606 283    2     34  11         Male  No      Yes    
2    2  106.025  6645 483    3     82  15        Female Yes     Yes    
3    3  104.593  7075 514    4     71  11         Male  No      No     
4    4  148.924  9504 681    3     36  11        Female No      No     
5    5   55.882  4897 357    2     68  16         Male  No      Yes    
6    6   80.180  8047 569    4     77  10         Male  No      No     
7    7   20.996  3388 259    2     37  12        Female No      No     
8    8   71.408  7114 512    2     87   9         Male  No      No     
9    9   15.125  3300 266    5     66  13        Female No      No     
10  10   71.061  6819 491    3     41  19        Female Yes     Yes    
11  11   63.095  8117 589    4     30  14         Male  No      Yes    
12  12   15.045  1311 138    3     64  16         Male  No      No     
13  13   80.616  5308 394    1     57   7        Female No      Yes    
14  14   43.682  6922 511    1     49   9         Male  No      Yes    
15  15   19.144  3291 269    2     75  13        Female No      No     
16  16   20.089  2525 200    3     57  15        Female No      Yes    
17  17   53.598  3714 286    3     73  17        Female No      Yes    
18  18   36.496  4378 339    3     69  15        Female No      Yes    
19  19   49.570  6384 448    1     28   9        Female No      Yes    
20  20   42.079  6626 479    2     44   9         Male  No      No     
21  21   17.700  2860 235    4     63  16        Female No      No     
22  22   37.348  6378 458    1     72  17        Female No      No     
23  23   20.103  2631 213    3     61  10         Male  No      Yes    
24  24   64.027  5179 398    5     48   8         Male  No      Yes    
25  25   10.742  1757 156    3     57  15        Female No      No     
26  26   14.090  4323 326    5     25  16        Female No      Yes    
27  27   42.471  3625 289    6     44  12        Female Yes     No     
28  28   32.793  4534 333    2     44  16         Male  No      No     
29  29  186.634 13414 949    2     41  14        Female No      Yes    
30  30   26.813  5611 411    4     55  16        Female No      No     
⋮   ⋮   ⋮       ⋮     ⋮      ⋮     ⋮   ⋮         ⋮      ⋮       ⋮      
371 371  35.610  6135 466    4     40  12         Male  No      No     
372 372  39.116  2150 173    4     75  15         Male  No      No     
373 373  19.782  3782 293    2     46  16        Female Yes     No     
374 374  55.412  5354 383    2     37  16        Female Yes     Yes    
375 375  29.400  4840 368    3     76  18        Female No      Yes    
376 376  20.974  5673 413    5     44  16        Female No      Yes    
377 377  87.625  7167 515    2     46  10        Female No      No     
378 378  28.144  1567 142    3     51  10         Male  No      Yes    
379 379  19.349  4941 366    1     33  19         Male  No      Yes    
380 380  53.308  2860 214    1     84  10         Male  No      Yes    
381 381 115.123  7760 538    3     83  14        Female No      No     
382 382 101.788  8029 574    2     84  11         Male  No      Yes    
383 383  24.824  5495 409    1     33   9         Male  Yes     No     
384 384  14.292  3274 282    9     64   9         Male  No      Yes    
385 385  20.088  1870 180    3     76  16         Male  No      No     
386 386  26.400  5640 398    3     58  15        Female No      No     
387 387  19.253  3683 287    4     57  10         Male  No      No     
388 388  16.529  1357 126    3     62   9         Male  No      No     
389 389  37.878  6827 482    2     80  13        Female No      No     
390 390  83.948  7100 503    2     44  18         Male  No      No     
391 391 135.118 10578 747    3     81  15        Female No      Yes    
392 392  73.327  6555 472    2     43  15        Female No      No     
393 393  25.974  2308 196    2     24  10         Male  No      No     
394 394  17.316  1335 138    2     65  13   

First, start with the simple model

$$y=\beta_0 + \epsilon$$

In [3]:
lm.fit=lm(Balance~1) # estimate a linear model regressing Balance on just the intercept
summary(lm.fit) # display results


Call:
lm(formula = Balance ~ 1)

Residuals:
    Min      1Q  Median      3Q     Max 
-520.01 -451.26  -60.51  342.99 1478.98 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   520.01      22.99   22.62   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 459.8 on 399 degrees of freedom


Note that this is the same as taking the unconditional mean of Balance.

In [4]:
mean(Balance)

[1] 520.015

So the average balance in the sample is $520.

Now, lets include a categorical Gender variable

$$ y= \beta_0 + \beta_1 x $$

where 

$$x = \begin{cases}
1, \text { if female} \\
0, \text { if male}\\
\end{cases}$$

In [15]:
lm.fit=lm(Balance~1 + Gender) # estimate a linear model regressing Balance on intercaept and gender
summary(lm.fit) # display results


Call:
lm(formula = Balance ~ 1 + Gender)

Residuals:
    Min      1Q  Median      3Q     Max 
-529.54 -455.35  -60.17  334.71 1489.20 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)    509.80      33.13  15.389   <2e-16 ***
GenderFemale    19.73      46.05   0.429    0.669    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 460.2 on 398 degrees of freedom
Multiple R-squared:  0.0004611,	Adjusted R-squared:  -0.00205 
F-statistic: 0.1836 on 1 and 398 DF,  p-value: 0.6685


How do we interpret thsese coeffecients?

$\hat\beta_0$=$509.80 is the average balance conditional on not being female (i.e. male)

$\beta_1$=$19.73 is the average **additional** balance conditional on being female

In [31]:
contrasts(Gender) # this funciton tells you how the lm() function coded the dummy variable

Female
 Male  0     
Female 1

We can verify that this regression is just taking the conditional means

In [5]:
isFemale <- Gender=="Female" # create a "logical" dummy variable for female

mean(Balance[isFemale]) # take average of Balance conditional on female

[1] 529.5362

In [6]:
mean(Balance[!isFemale]) # take average of Balance conditional on male

[1] 509.8031